x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -154,5 +156,106 @@ export function checksFactory(
     };
   }
 
-  return { checkStatus };
+  async function deleteJobsCheck(
+    request: KibanaRequest,
+    jobType: JobType,
+    jobIds: string[],
+    spacesEnabled: boolean,
+    resolveMlCapabilities: ResolveMlCapabilities
+  ) {
+    if (jobType !== 'anomaly-detector' && jobType !== 'data-frame-analytics') {
+      throw Boom.badRequest('Job type must be "anomaly-detector" or "data-frame-analytics"');
+    }
+
+    const mlCapabilities = await resolveMlCapabilities(request);
+    if (mlCapabilities === null) {
+      throw Boom.internal('mlCapabilities is not defined');
+    }
+
+    if (
+      mlCapabilities.canDeleteJob === false ||
```

Comment:
Is it possible that the user could have one permission and not the other? Or is job deletion a single permission across ML?

Reply:
at the moment no, we only have ML viewer or ML admin roles.
But just in case we ever allow fine grained capabilities i'll change this check to be more explicit.

Reply:
Changed in https://github.com/elastic/kibana/pull/83871/commits/5951246541eee2994409fece49af41a4e74a1c71